app/sidekiq/evss/disability_compensation_form/form0781_document_upload_failure_email.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require 'va_notify/service'
+
+module EVSS
+  module DisabilityCompensationForm
+    class Form0781DocumentUploadFailureEmail < Job
+      STATSD_METRIC_PREFIX = 'api.form_526.veteran_notifications.form0781_upload_failure_email'
+
+      # retry for one day
+      sidekiq_options retry: 14
+
+      sidekiq_retries_exhausted do |msg, _ex|
+        job_id = msg['jid']
+        error_class = msg['error_class']
+        error_message = msg['error_message']
+        timestamp = Time.now.utc
+        form526_submission_id = msg['args'].first
+
+        # Job status records are upserted in the JobTracker module
+        # when the retryable_error_handler is called
+        form_job_status = Form526JobStatus.find_by(job_id:)
+        bgjob_errors = form_job_status.bgjob_errors || {}
+        new_error = {
+          "#{timestamp.to_i}": {
+            caller_method: __method__.to_s,
+            timestamp:,
+            form526_submission_id:
+          }
+        }
+        form_job_status.update(
+          status: Form526JobStatus::STATUS[:exhausted],
+          bgjob_errors: bgjob_errors.merge(new_error)
+        )
+
+        Rails.logger.warn(
+          'Form0781DocumentUploadFailureEmail retries exhausted',
+          {
+            job_id:,
+            timestamp:,
+            form526_submission_id:,
+            error_class:,
+            error_message:
+          }
+        )
+
+        StatsD.increment("#{STATSD_METRIC_PREFIX}.exhausted")
+      rescue => e
+        Rails.logger.error(
+          'Failure in Form0781DocumentUploadFailureEmail#sidekiq_retries_exhausted',
+          {
+            job_id:,
+            messaged_content: e.message,
+            submission_id: form526_submission_id,
+            pre_exhaustion_failure: {
+              error_class:,
+              error_message:
+            }
+          }
+        )
+        raise e
+      end
+
+      def perform(form526_submission_id)
+        form526_submission = Form526Submission.find(form526_submission_id)
+
+        with_tracking('Form0781DocumentUploadFailureEmail', form526_submission.saved_claim_id, form526_submission_id) do
+          @notify_client ||= VaNotify::Service.new(Settings.vanotify.services.benefits_disability.api_key)
```

Comment:
memoizing doesn't do anything here. the `||=` is for when you have a method that memoizes for you, which could be a nice refactor. ALso i think code climate had something to say about your method chain.  I would just make it a normal variable and remove the `||`

Reply:
Oh good catch, yeah my bad copy pasted this from older mailers I should have taken a closer look. Fixed in https://github.com/department-of-veterans-affairs/vets-api/pull/18206/commits/0cf08174cc0c83ed00ef564cbdde1157c7aeb11e